specs/NonClientHitTestKind.md

```diff
@@ -0,0 +1,203 @@
+    CHECK_FAILURE(CompositionController->SendMouseInput(
+        static_cast<COREWEBVIEW2_MOUSE_EVENT_KIND>(message),
+        static_cast<COREWEBVIEW2_MOUSE_EVENT_VIRTUAL_KEYS>(GET_KEYSTATE_WPARAM(wParam)),
+        mouseData, point));
+    return 0;
+    }
+    
+}
+// The following code forwards mouse messages to the WebView2 - End
+
+switch(message) 
+{
+    ....
+    // Handling this message is important for enabling dragging
+    case WM_NCHITTEST:
+    POINT point {GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)};
+    COREWEBVIEW2_HIT_TEST_KIND result = NULL;
+    CompositionController->NonClientHitTestAtPoint(&point, &result);
+
+    if (result == 
+        COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT::COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_HTCAPTION)
+    {
+        return HTCAPTION;
+    }
+
+    break;
+    ....
+}
+....
+}
+```
+#### C#/WinRT
+```c#
+protected override void WndProc(ref Message m)
+{
+....
+
+bool isCapturingMouse = false;
+
+if (m.Msg == WM_NCHITTEST)
+{
+    Point point = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+    if (CompositionController.NonClientHitTestAtPoint(point) == 
+        CoreWebView2NonClientHitTestResult.Htcaption) 
+    {
+        m.Result = (IntPtr)HTCAPTION;
+    }
+}
+// The following code forwards mouse messages to the WebView2 - Start
+else if (m.msg >= WM_MOUSEFIRST && m.msg <= WM_MOUSELAST) 
+{
+  
+    Point point = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+    // IsPointInWebview is an example of a function which your app can implement that checks if 
+    // the mouse message point lies inside the rect of the visual that hosts the WebView
+    if (IsPointInWebView(point) || m.msg == WM_MOUSELEAVE || isCapturingMouse) 
+    {
+
+        if (m.msg == WM_NCRBUTTONDOWN && m.w_param == HTCAPTION) 
+        {
+            // capturing the mouse will allow us to begin forwarding messages 
+            // to webview
+            isCapturingMouse = true;
+            SetCapture(mainWindowHwnd);
+        }
+        else if (message ==  WM_RBUTTONUP && GetCapture() == mainWindowHwnd) 
+        {
+            // no longer need to capture the mouse for draggbale regions support 
+            // after this
+            isCapturingMouse = false;
+            ReleaseCapture();
+        }
+        
+        // forward mouse messages to WV
+        CompositionController.SendMouseInput(
+            (CoreWebView2MouseEventKind)message,
+            (CoreWebView2MouseEventVirtualKeys)GET_KEYSTATE_WPARAM(wParam),
+            mouseData,
+            point);
+
+        m.Result = 0
+        return;
+    }
+        
+}
+// The following code forwards mouse messages to the WebView2 - Start
+    ....
+}
+       
+```
+# API Details
+## Win32 C++
+```cpp
+/// This enum contains values representing possible regions that Webview
+/// can support
+typedef enum COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT {
+    /// Caption region
+    COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_CAPTION,
```

Comment:
The other visual hosting input type enums (COREWEBVIEW2_POINTER_EVENT_KIND, COREWEBVIEW2_MOUSE_EVENT_VIRTUAL_KEYS, COREWEBVIEW2_MOUSE_EVENT_KIND) all have values that match their Win32 counterpart. Should we set these to match? (@david-risney)
(i.e. COREWEBVIEW2_NON_CLIENT_HIT_TEST_RESULT_CAPTION = 2)

Reply:
Yes that sounds convenient.

Reply:
will do